src/main/scala/mesosphere/marathon/core/appinfo/impl/AppInfoBaseData.scala

```diff
@@ -122,11 +123,11 @@ class AppInfoBaseData(
 
     lazy val enrichedTasksFuture: Future[Seq[EnrichedTask]] = {
       def statusesToEnrichedTasks(
-        tasksById: Map[String, MarathonTask],
+        tasksById: Map[Task.Id, MarathonTask],
         statuses: Map[String, collection.Seq[Health]]): Seq[EnrichedTask] = {
         for {
           (taskId, healthResults) <- statuses.to[Seq]
-          task <- tasksById.get(taskId)
+          task <- tasksById.get(Task.Id(taskId))
```

Comment:
The HealthCheckManager could give us a TaskId instead of a String.
If not done in this PR: can we somehow mark those occurrences? (It will be hard to identify all places afterwards)